flocker/node/agents/ebs.py

```diff
@@ -243,59 +335,88 @@ def _blockdevicevolume_from_ebs_volume(ebs_volume):
     )
 
 
-def _wait_for_volume(volume,
-                     start_status,
-                     transient_status,
-                     end_status):
+def _get_ebs_volume_state(volume):
+    """
+    Fetch input EBS volume's latest state from backend.
+
+    :param boto.ec2.volume volume: Volume that needs state update.
+
+    :returns: EBS volume with latest state known to backend.
+    :rtype: boto.ec2.volume
+
+    """
+    return volume.update()
+
+
+def _wait_for_volume_state_change(operation,
+                                  volume,
+                                  update=_get_ebs_volume_state,
+                                  timeout=VOLUME_STATE_CHANGE_TIMEOUT):
```

Comment:
The downside to this approach is that you then need another test or two for `should_finish` integration with `wait`. Tradeoffs everywhere!